Classes/Units/Controllers/Inventory.js

```diff
@@ -65,13 +65,42 @@ class Inventory {
 		if(slot === false)
 			return false;
 
+		var effectiveGoldCost = Item.GoldCost
+
+		if(Item.from)
+		{
+			// If an Item can be build from another items
+			// set the effective gold Cost to substract
+			// Meanwhile remove the "from" items
+			// At the end, reassign the item slot
+			effectiveGoldCost = this.buildFromItems( Item )
```

Comment:
It's just my fancy to not adding whitespaces at the edges of the parentheses so I will allow it,
but please use semicolons `;` at the end of expression.